### PR TITLE
Issue 52: RMSE & MAE

### PR DESCRIPTION
**Pull request by sheetalsattiraju**
## What does this PR do?
- This PR adds rmse() (Root Mean Square Error) and mae() (Mean Absolute Error) as metrics in kstats-core/src/commonMain/kotlin/org/oremif/kstats/descriptive/RegressionMetrics.kt.
- Each metric is implemented with three overloads (DoubleArray, Iterable<Double>, Sequence<Double>) and uses the project's summation helper (compensatedSum()).
- A corresponding test file is included in kstats-core/src/commonTest/kotlin/org/oremif/kstats/descriptive/RegressionMetricsTest.kt, mirroring the structure of CentralMomentTest.kt.

## Why was this PR needed?
- In issue #52, the maintainer asked for these metrics to be added to kstats-core.
-  RMSE and MAE are standard tools for evaluating prediction quality in regression, recommendation systems, and forecasting

## What are the relevant issue numbers?
- Issue #52 

## Does this PR meet the acceptance criteria?
- [x] Tests added for new/changed behavior in kstats-core/src/commonTest/kotlin/org/oremif/kstats/descriptive/RegressionMetricsTest.kt covering known values, edge cases, invalid input, NaN propagation, and overload consistency (24 test cases total)
- [x] All tests passing
- [x] Follows project style guide — implementation mirrors the structure in existing CentralTendency.kt file
- [x] No breaking changes introduced — 2 files added, no existing metrics or APIs were modified

### CODE REVIEW

File: kstats-core/src/commonMain/kotlin/org/oremif/kstats/descriptive/RegressionMetrics.kt [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+ *
+ * ### Example:
+ * ```kotlin
+ * val actual = doubleArrayOf(3.0, 5.0, 2.5, 7.0)
+ * val predicted = doubleArrayOf(2.8, 5.2, 2.1, 6.8)
+ * rmse(actual, predicted) // 0.2646...
+ * ```
+ *
+ * @return the root mean square error between actual and predicted.
+ * Version 1: DoubleArray
```
Comment (devcrocod) [inline kstats-core/src/commonMain/kotlin/org/oremif/kstats/descriptive/RegressionMetrics.kt]: Why need this comment here after the return?

File: kstats-core/src/commonMain/kotlin/org/oremif/kstats/descriptive/RegressionMetrics.kt [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    val squaredDiffs = DoubleArray(actual.size) {i ->
+        val diff = actual[i] - predicted[i]
+        diff * diff
+    }
+    return sqrt(squaredDiffs.compensatedSum() / actual.size)
+}
+
+/**
+ * Computes the Root Mean Square Error (RMSE) between actual and predicted values.
+ * Version 2: Iterable<Double> (for list, sets, etc)
```
Comment (devcrocod) [inline kstats-core/src/commonMain/kotlin/org/oremif/kstats/descriptive/RegressionMetrics.kt]: This comment is useless for users
Specifying the version provides no useful information. It doesn't matter if it is version 1 or 2.
Comments about parameters/arguments are much more useful. That is exactly what is missing here.

File: kstats-core/src/commonTest/kotlin/org/oremif/kstats/descriptive/RegressionMetricsTest.kt
Comment (devcrocod) [inline kstats-core/src/commonTest/kotlin/org/oremif/kstats/descriptive/RegressionMetricsTest.kt:1]: I don't see any tests for infinity, could you please add them?

File: kstats-core/src/commonMain/kotlin/org/oremif/kstats/descriptive/RegressionMetrics.kt [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+ */
+ 
+public fun rmse(actual: Iterable<Double>, predicted: Iterable<Double>): Double {
```
Comment (devcrocod) [inline kstats-core/src/commonMain/kotlin/org/oremif/kstats/descriptive/RegressionMetrics.kt:49]: This will result in NaN overflow.
It should be infinity

**Review by devcrocod — CHANGES_REQUESTED** [review]
please use apiDump to update the api file
I left a few comments, please take a look at them

**Comment by sheetalsattiraju** [comment]
@devcrocod Thank you for the comments!

I removed the unnecessary comments and replaced them with parameter explanations and updated the api file. I also added the infinity tests as requested and ran into one edge case. The existing DoubleArray.compensatedSum() returns NaN when the input contains infinity, which causes the rmse/mae DoubleArray infinity tests to fail. The Iterable implementations handle the infinity case correctly now, as confirmed by the new tests.

I didn't want to modify the existing MathUtils.kt implementation without checking first, since this may be related to how compensatedSum() is intended to handle infinite values. Could you take a look and let me know if this behavior is expected, or if compensatedSum() should handle infinities differently?

**Review by devcrocod — APPROVED** [review]
lgtm
thanks for contributing!
